helpers/useSession.tsx

```diff
@@ -1,23 +1,37 @@
 import useSWR from 'swr'
 import { User } from './models/User'
+import _ from 'lodash'
 
 const SERVER_URL = process.env.SERVER_URL
 
 export const fetcher = (url: string) =>
-  fetch(url, { credentials: 'include' }).then(r => r.json())
+  fetch(url, {
+    method: 'POST',
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      query: `
+            query {
+              session {
+                user {
+                    username
+                }
+              }
+            }
+          `
+    })
+  }).then(r => r.json())
 
 export type SessionData = {
-  data: {
-    userInfo?: User
-    errorMessage?: string
-    success: boolean
-  }
-  error?: boolean
+  session: User | null
 }
 
 const useSession = (): SessionData => {
-  const { data, error } = useSWR(`${SERVER_URL}/session`, fetcher)
-  return { data, error }
+  const { data } = useSWR(`${SERVER_URL}`, fetcher)
```

Comment:
Not a big deal but `SERVER_URL` it's already a string you do not need to use string interpolation